pkg/controller/plan/adapter/openstack/client.go

```diff
@@ -272,12 +274,13 @@ func (r *Client) PreTransferActions(vmRef ref.Ref) (ready bool, err error) {
 			r.Log.Info("the image properties are in sync, cleaning the image",
 				"vm", vm.Name, "image", inventoryImage.Name, "properties", inventoryImage.Properties)
 			originalVolumeID := inventoryImage.Properties[forkliftPropertyOriginalVolumeID].(string)
-			err = r.cleanup(vm, originalVolumeID)
-			if err != nil {
-				r.Log.Error(err, "cleaning the image",
-					"vm", vm.Name, "image", image.Name)
-				return
-			}
+			go func() {
+				err := r.cleanup(vm, originalVolumeID)
+				if err != nil {
+					r.Log.Error(err, "failed to cleanup")
+				}
+			}()
```

Comment:
@bennyz can you please remind me why we moved this into a goroutine? is it an optimization that is meant to reduce the execution time?

Reply:
mainly because it's a best effort operation which we don't need to wait for

Reply:
ah is it a blocking operation?

Reply:
yes, specifically the waiting part

Reply:
hehe, oops, I forgot about the rest of this PR as I'm concentrated on #491 ..